[quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36844 [quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops**
* #36818 [quant][graph] Support for quantized::mul and quantized::mul_scalar
* #36817 [quant][graph] Update quantize_dynamic_script API to take sample model args

Summary:

Test Plan:
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_relu
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_scalar_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21134440](https://our.internmc.facebook.com/intern/diff/D21134440)

supriyar committed:
[quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops
Summary:

Test Plan:
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_relu
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_scalar_relu
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -717,6 +717,27 @@ graph(%self, %input, %inplace):
+     %second_output = prim::CallMethod[name="forward"](%second_module, %first_output)
+     return (%second_output) )");
+
+  const PatternInfo mul_functional_relu = PatternInfo::parse_from_str(R"(
```
do we have inplace_mul_functional_relu as well?

supriyar replied:
Yes, let me add that here

supriyar committed:
Update on "[quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops"


Summary:

Test Plan:
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_relu
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_scalar_relu
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graph] Add quantized::mul_relu and quantized::mul_scalar_relu ops"


Summary:

Test Plan:
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_relu
python test/quantization/test_quantize_script.py TestQuantizeScriptPTSQOps.test_quantized_mul_scalar_relu
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -725,6 +754,10 @@ graph(%self, %input, %inplace):
+          mul_module_relu,
+          inplace_mul_module_relu,
+          mul_functional_relu,
+          inplace_mul_functional_relu,
```
I saw we only have two entries for `add_relu` right now, could you add another PR to add two more entries for `add_relu` as well?
